docs/nddata/index.rst

```diff
@@ -7,126 +7,142 @@ N-dimensional datasets (`astropy.nddata`)
+  additional functionality like an reading/writing, simple arithmetic
+  operations and slicing.
++ `~astropy.nddata.StdDevUncertainty` a class that can store and propagate
+  uncertainties for a NDData object.
++ General utility functions (:ref:`nddata_utils`) for operations on these
+  classes, including a decorator to facilitate writing functions for such
+  classes.
++ A framework including different mixins and metaclasses to facilitate
+  customizing subclasses.
 
 Getting started
 ===============
 
-Of the classes provided by `~astropy.nddata`, the place to start for most
-users will be `~astropy.nddata.NDData`, which by default uses a numpy array to
-store the data. Designers of new classes should also look at
-`~astropy.nddata.NDDataBase` before deciding what to subclass from.
-
 NDData
 ------
 
 The primary purpose of `~astropy.nddata.NDData` is to act as a *container* for
 data, metadata, and other related information like a mask.
 
 An `~astropy.nddata.NDData` object can be instantiated by passing it an
-n-dimensional Numpy array::
+n-dimensional `numpy` array::
 
     >>> import numpy as np
     >>> from astropy.nddata import NDData
     >>> array = np.zeros((12, 12, 12))  # a 3-dimensional array with all zeros
-    >>> ndd = NDData(array)
+    >>> ndd1 = NDData(array)
 
-or something that can be converted to an array::
+or something that can be converted to an `numpy.ndarray`::
 
     >>> ndd2 = NDData([1, 2, 3, 4])
+    >>> ndd2
+    NDData([1, 2, 3, 4])
 
-It is also possible to initialize `~astropy.nddata.NDData` with more exotic
-objects; see :ref:`nddata_details` for more information.
-
-The underlying Numpy array can be accessed via the ``data`` attribute::
+and can be accessed again via the ``data`` attribute::
 
-    >>> ndd.data
-    array([[[ 0., 0., 0., ...
-    ...
+    >>> ndd2.data
+    array([1, 2, 3, 4])
 
-Values can be masked using the ``mask`` attribute::
+and it supports additional properties like a ``unit`` or ``mask`` for the data,
+a ``wcs`` (world coordinate system) and ``uncertainty`` of the data and
+additional ``meta`` attributes:
 
-     >>> ndd_masked = NDData(ndd, mask = ndd.data > 0.9)
+    >>> data = np.array([1,2,3,4])
+    >>> mask = data > 2
+    >>> unit = 'erg / s'
+    >>> from astropy.nddata import StdDevUncertainty
+    >>> uncertainty = StdDevUncertainty(np.sqrt(data)) # representing standard deviation
+    >>> meta = {'object': 'fictional data.'}
+    >>> from astropy.coordinates import SkyCoord
+    >>> wcs = SkyCoord('00h42m44.3s', '+41d16m09s')
+    >>> ndd = NDData(data, mask=mask, unit=unit, uncertainty=uncertainty,
+    ...              meta=meta, wcs=wcs)
+    >>> ndd
+    NDData([1, 2, 3, 4])
 
-A mask value of `True` indicates a value that should be ignored, while a mask
-value of `False` indicates a valid value.
+the representation does not show additional attributes but these can be
+accessed like ``data`` above::
```

Comment:
Sure, but basically the solution for now is to access the attributes directly....how about just mentioning that other attributes need to be accessed directly?